jetkit/db/extid.py

```diff
@@ -16,17 +18,33 @@ class ExtID(Generic[T]):
 
     # UUID that can be used for semi-secret key
     extid = Column(
-        UUID(as_uuid=True), nullable=False, server_default=text("uuid_generate_v4()")
+        UUID(as_uuid=True),
+        nullable=False,
+        server_default=text("uuid_generate_v4()"),
+        index=True,
     )
 
     @classmethod
-    def get_by_extid(cls: T, uuid: str) -> Optional[T]:
-        return cls.query.filter_by(extid=uuid).one_or_none()
+    def get_by_extid(cls: Type[db.Model], uuid: str) -> Optional[T]:
+        try:
+            return cls.query.filter_by(extid=uuid).one_or_none()
+        except DataError:  # This means that extid is not a valid uuid
```

Comment:
heh? not valid you mean resource with that type is not in database.

Reply:
`ERROR:  invalid input syntax for type uuid: "foo"`